File: spara27/lib/src/models/ahorro_model.dart
import 'dart:convert';

Ahorro ahorroFromJson(String str) => Ahorro.fromJson(json.decode(str));

String ahorroToJson(Ahorro data) => json.encode(data.toJson());

class Ahorro {
  Ahorro({
    this.uid,
    this.monto,
    this.descripcion,
    this.fechaInicio,
  });

  String? uid;
  double? monto;
  String? descripcion;
  DateTime? fechaInicio;

  factory Ahorro.fromJson(Map<String, dynamic> json) => Ahorro(
        uid: json["uid"],
        monto: json["monto"].toDouble(),
        descripcion: json["descripcion"],
        fechaInicio: DateTime.parse(json["fechaInicio"]),
      );
  factory Ahorro.created() =>
      Ahorro(uid: "", monto: 0, descripcion: "", fechaInicio: DateTime.now());
  Map<String, dynamic> toJson() => {
        "uid": uid,
        "monto": monto,
        "descripcion": descripcion,
        "fechaInicio": fechaInicio!.toIso8601String(),
      };
}

File: functions/src/models/ahorro-model.ts
export interface Ahorro {
    idahorro?:   String;
    monto:       number;
    metaAhorro:  number;
    descripcion: string;
    fechaInicio: string;
    fechaFinal:  string;
}
export function Ahorro(data :any, id?:string){
    const { monto, metaAhorro, descripcion, fechaInicio, fechaFinal } = data;

    let object :Ahorro = { 
    idahorro: id,               
    monto:   monto,
    metaAhorro:  metaAhorro,
    descripcion:descripcion,
    fechaInicio: fechaInicio,
    fechaFinal:  fechaFinal,
    };
    return object;
}